### PR TITLE
Added documentation for adding custom fields to registration with Spring Boot.

### LINKED ISSUE #420: Add documentation to spring boot about customizing views and adding additional fields for account registration
Feedback from a enterprise customer on a call.

He wants step by step guide to how to add additional fields to the registration form.

### PR DESCRIPTION
**Pull request by mraible**
Fixes #420 and #886.

#420 also requests documentation for customizing views, but that seems to exist already.

https://docs.stormpath.com/java/spring-boot-web/views.html#views

### CODE REVIEW

File: examples/servlet/src/main/resources/stormpath.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+stormpath.web.register.form.fields.birthday.type = text
+stormpath.web.register.form.fields.birthday.label = Birthday
+stormpath.web.register.form.fields.birthday.enabled = true
+stormpath.web.register.form.fields.birthday.visible = true
+stormpath.web.register.form.fields.birthday.required = true
+stormpath.web.register.form.fields.birthday.placeholder = Birthday
```
Comment (mzumbadoate) [inline examples/servlet/src/main/resources/stormpath.properties]: do we want to keep these changes?

Comment (mraible) [inline examples/servlet/src/main/resources/stormpath.properties]: Nope! Good catch.

File: extensions/spring/boot/docs/source/registration.rst [hunk 2]
```diff
@@ -112,6 +112,27 @@ You can customize this list with optional directives as necessary, but note:
 
 .. _password strength:
 
+Custom Data
+^^^^^^^^^^^
+
+You can capture `Custom Data`_ fields when users register by adding configuration properties. For example,
```
Comment (mrioan) [inline extensions/spring/boot/docs/source/registration.rst]: `The registration form provides the ability to have custom fields defined by the developer. The registration controller will automatically figure out which are those fields (i.e not part of the pre-defined properties supported by the `Account`) and they will be added as `Custom Data` in the account. 

For example, let's suppose we want to add a custom field to capture a user's birthday during registration. Then the following properties should be added to the `stormpath.properties` file:

File: extensions/spring/boot/docs/source/registration.rst [hunk 3]
```diff
@@ -112,6 +112,27 @@ You can customize this list with optional directives as necessary, but note:
...
+    stormpath.web.register.form.fields.birthday.visible = true
+    stormpath.web.register.form.fields.birthday.required = true
+    stormpath.web.register.form.fields.birthday.placeholder = Birthday
+
+When the registration form is rendered, this field will be added to the bottom of the form.
+
+   .. image:: /_static/register-with-birthday.png
+
+When the form is submitted, the field's name and value will be added to the account's custom data.
+
```
Comment (mrioan) [inline extensions/spring/boot/docs/source/registration.rst]: But if you are looking for a true internationalization experience then you should define the `label` property like this:
.. code-block:: properties

    stormpath.web.register.form.fields.birthday.label = stormpath.web.register.form.fields.birthday.label

And then, in your `i18n_en.properties` file you would add:

.. code-block:: properties

    stormpath.web.register.form.fields.birthday.label = Birthday

While in your tentative `i18n_es.properties` file you would add:

.. code-block:: properties

    stormpath.web.register.form.fields.birthday.label = Fecha de nacimiento

**Comment by mrioan** [comment]
@mraible besides addressing my comments please also add the same documentation to the Servlet plugin. Back to you. Thanks

**Comment by mrioan** [comment]
@mraible please be aware that this section of the Boot documentation needs to be updated: `https://docs.stormpath.com/java/spring-boot-web/registration.html#form-fields`:

**Comment by mraible** [comment]
@mrioan Updated based on your comments. Please review again.

**Comment by mraible** [comment]
@mrioan I think @mzumbadoate fixed the form fields stuff in https://github.com/stormpath/stormpath-sdk-java/pull/867.

**Comment by mrioan** [comment]
@mraible LGTM. Feel free to merge. Thanks!

**Pull request merged**